code/modules/research/nanites/nanite_programs/utility.dm

```diff
@@ -213,6 +213,53 @@
 	rogue_types = list(/datum/nanite_program/skin_decay)
 	var/access = list()
 
+/datum/nanite_program/research
+	name = "Distributed Computing"
+	desc = "The nanites aid the research servers by performing a portion of its calculations, increasing research point generation."
+	use_rate = 0.2
+	rogue_types = list(/datum/nanite_program/toxic)
+
+/datum/nanite_program/research/active_effect()
+	if(!iscarbon(host_mob))
+		return
+	var/points = 1
+	if(!host_mob.client) //less brainpower
+		points *= 0.25
+	SSresearch.science_tech.add_point_list(list(TECHWEB_POINT_TYPE_GENERIC = points))
+
+/datum/nanite_program/researchplus
+	name = "Neural Network"
+	desc = "The nanites link the host's brains together forming a neural research network, that becomes more efficient with the amount of total hosts."
+	use_rate = 0.3
+	rogue_types = list(/datum/nanite_program/brain_decay)
+
+/datum/nanite_program/researchplus/enable_passive_effect()
+	. = ..()
+	if(!iscarbon(host_mob))
+		return
+	if(host_mob.client)
+		SSnanites.neural_network_count++
+	else
+		SSnanites.neural_network_count += 0.25
+
+/datum/nanite_program/researchplus/disable_passive_effect()
+	. = ..()
+	if(!iscarbon(host_mob))
+		return
+	if(host_mob.client)
+		SSnanites.neural_network_count--
+	else
+		SSnanites.neural_network_count -= 0.25
```

Comment:
You have to somehow store the gain to remove that, currently someone can:

- Activate it, increesing the value by 1
- Log off
- Ask someone to deactivate while logged off, decreasing by 0.25
- Rince and repeat for free point gain